pkg/packetbrokeragent/agent.go

```diff
@@ -303,7 +308,7 @@ func New(c *component.Component, conf *Config, opts ...Option) (*Agent, error) {
 			messageEncrypter:    a,
 			contextDecoupler:    a,
 			tenantIDExtractor:   a.tenantIDExtractor,
-			frequencyPlansStore: a.FrequencyPlans,
+			frequencyPlansStore: fps,
```

Comment:
This needs to become a function that takes a `context.Context` and returns `(frequencyPlanStore, error)`. The context here won't contain the tenant ID.